source-hubspot-native/source_hubspot_native/api.py

```diff
@@ -271,14 +460,260 @@ async def fetch_recent_tickets(
     log: Logger, http: HTTPSession, since: datetime, cursor: PageCursor
 ) -> tuple[Iterable[tuple[datetime, str]], PageCursor]:
 
-    url = f"{HUB}/crm-objects/v1/change-log/tickets"
-    params = {"timestamp": int(since.timestamp() * 1000) - 1}
+    url = f"{HUB}/crm/v3/objects/tickets"
+    
+    result = OldRecentTicket.model_validate_json(
+        await http.request(log, url, params=None)
+    )
+
+    return (
+        (r.updatedAt, str(r.id))
+        for r in result.results
+    ), None and None
+
+async def fetch_recent_contacts_lists(
+    log: Logger, http: HTTPSession, since: datetime, page: PageCursor
+) -> tuple[Iterable[tuple[datetime, str, Any]], PageCursor]:
+
+    url = f"{HUB}/contacts/v1/lists"
+    params = {"count": 100, "offset": page} if page else {"count": 250}
+
+    data = await http.request(log, url, params=params)
+
+
+    result = V1CustomCRMObject2.model_validate_json(
+        data
+    )
+
+    for item in result.lists:
+        item['listId'] = str(item['listId'])
+
+    if result.hasMore is False:
+        return (
+            (_ms_to_dt(r['updatedAt']), str(r['listId']), r)
+            for r in result.lists if _ms_to_dt(r['updatedAt']) > since
```

Comment:
This filtering conceptually makes since to me...since we want things after `since`, only emit documents that come after `since`. But I don't know why we don't have that on others, like `fetch_recent_contacts`? It seems they are doing pretty much the same thing. How does that work for `fetch_recent_contacts`, but it wouldn't work here?

Reply:
This is a really clever idea from johnny, were we call this [endpoint](https://legacydocs.hubspot.com/docs/methods/lists/get_list_contacts_recent)  where it formats the results in order of more recent to less recent. Then, the `fetch_recent_contacts` method gets when this data has been updated and its Id and sents it to the `fetch_changes` function. This function is responsible to filter these timestamps with our `cut_off` / `since` value, and the ones that passes are then collected using batch functions